tests/components/blebox/test_helpers.py

```diff
@@ -0,0 +1,37 @@
+"""Blebox helpers tests."""
+
+from aiohttp.helpers import BasicAuth
+
+from homeassistant.components.blebox.helpers import get_maybe_authenticated_session
+from homeassistant.core import HomeAssistant
+
+DICT_SAMPLE = {"a": "", "b": "value"}
+DICT_TEST_PARAMETERS = [("a", None), ("b", "value")]
+
+USERNAME = "user"
+PASSWORD = "password"
+
+SESSION_TEST_PARAMS = [
+    ("", "", None),
+    ("user", "password", BasicAuth),
+]
+
+
+async def test_get_maybe_authenticated_session_none(
+    hass: HomeAssistant, param=SESSION_TEST_PARAMS[0]
+):
+    """Tests if session auth is None."""
+    session = get_maybe_authenticated_session(
+        hass=hass, username=param[0], password=param[1]
+    )
+    assert session.auth is param[2]
+
+
+async def test_get_maybe_authenticated_session_auth(
+    hass: HomeAssistant, param=SESSION_TEST_PARAMS[1]
+):
+    """Tests if session have BasicAuth."""
+    session = get_maybe_authenticated_session(
+        hass=hass, username=param[0], password=param[1]
+    )
+    assert isinstance(session.auth, param[2])
```

Comment:
Create one parametrize test for it

Reply:
I think this have to remain as is, as `isinstance` cannot be used against `None`, and `is` checks if it's the same object.

Reply:
Okay did not see that. But then please remove `SESSION_TEST_PARAMS` and replace all `params[x]` with the actual value